packages/core/src/stake/ensureTokenAccount.ts

```diff
@@ -5,9 +5,10 @@ export const ensureTokenAccount = async (
   transaction: Transaction,
   payer: PublicKey,
   stSolMint: PublicKey,
+  allowOwnerOffCurve: boolean = false,
 ) => {
   // Creating the associated token account if not already exist
-  const associatedStSolAccount = await getAssociatedTokenAddress(stSolMint, payer);
+  const associatedStSolAccount = await getAssociatedTokenAddress(stSolMint, payer, allowOwnerOffCurve);
```

Comment:
Can we extend here an error (as I understand error TokenOwnerOffCurveError comes from this part) using our custom class?

Reply:
No, we can't.

Reply:
Why not? Put it into try..catch
and then try to select exactly TokenOwnerOffCurveError errors

Reply:
I believe it's really important

Reply:
why it's really important? Just I think `TokenOwnerOffCurveError` more understandable for solana devs.

Reply:
In the other side, it could be useful (not really important), we can add it to be perfect.

Reply:
We should wrapp all known errors to let devs a general method how can they detect errors
based on type, that we add in our case

Reply:
```ts
  try {
    // Creating the associated token account if not already exist
    const associatedStSolAccount = await getAssociatedTokenAddress(stSolMint, payer, allowOwnerOffCurve);
    transaction.add(createAssociatedTokenAccountInstruction(payer, associatedStSolAccount, payer, stSolMint));

    return associatedStSolAccount;
  } catch (error) {
    if (error instanceof TokenInvalidAccountOwnerError) {
      throw new ErrorWrapper({ error, code: ERROR_CODE.PUBLIC_KEY_IS_PDA });
    }

    throw error;
  }
```